### DISCUSSION TITLE
Enhancement: Make the Docker image compatible for using Docker Secrets

### DISCUSSION THREAD

**Discussion by prononext** [danny-avila/LibreChat#3190] [Feature Requests & Suggestions]
### What features would you like to see added?

It would be really nice if the Docker Images would be compatible to Docker Secrets https://docs.docker.com/engine/swarm/secrets/

### More details

You could use the _FILE suffix on any ENV to encrypt sensible information like API Keys and other secrets inside Docker, so it is not available in clear inside the .env file. Every major image out there like for example mysql, postgres, mogo and many more are having this enabled on all Environment Variables. 

For a save setup without possibly exposed API-Keys and secrets that would be a secure way to go. 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by prononext** [comment]
Are there any plans to make the docker container compatible with docker-secrets?

**Comment by prononext** [comment]
Are there any changes on this topic?

**Comment by w41x** [comment]
This security concern remains critically important and should be prioritized.

The current implementation poses significant security risks by requiring sensitive credentials (such as `CREDS_KEY` and `MONGO_URI` containing passwords) to be passed directly as environment variables or stored in plaintext `.env` files. This approach contradicts security best practices and creates unnecessary attack vectors.

**Recommended improvements:**

1. **File-based secret injection**: Implement `*_FILE` variants for all sensitive variables (e.g., `CREDS_KEY_FILE`, `MONGO_PASSWORD_FILE`). This pattern, widely adopted by projects like Docker's official images, enables integration with secret management systems (Docker Secrets, Kubernetes Secrets, HashiCorp Vault, etc.).

2. **Decomposed database configuration**: Provide the option to specify database connection parameters separately:
   - `MONGO_HOST`
   - `MONGO_USER`
   - `MONGO_PASSWORD` / `MONGO_PASSWORD_FILE`
   - `MONGO_DATABASE`
   - Additional parameters as needed

This would allow users to leverage file-based secrets for credentials while keeping non-sensitive parameters as standard environment variables.

These changes would significantly improve LibreChat's security posture and make it production-ready for enterprise and security-conscious deployments.
